scripts/setup_linux.sh

```diff
@@ -1,61 +1,254 @@
+  _positional_names=('_arg_mjkey_path' )
+
+  for (( ii = 0; ii < ${#_positionals[@]}; ii++))
+  do
+    error_msg="$(echo "Error during argument parsing, possibly an Argbash" \
+      "bug.")"
+    eval "${_positional_names[ii]}=\${_positionals[ii]}" || \
+      die "${error_msg}" 1
+  done
+}
+
+parse_commandline "$@"
+handle_passed_args_count
+assign_positional_args
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+
+# Verify there's a file in the mjkey path
+test -f "${_arg_mjkey_path}" || _PRINT_HELP=yes die \
+  "The path ${_arg_mjkey_path} of the MuJoCo key is not valid." 1
+
+# Make sure that we're under the garage directory
+GARAGE_DIR="$(readlink -f $(dirname $0)/..)"
+cd "${GARAGE_DIR}"
+
+# File where environment variables are stored
+BASH_RC="${HOME}/.bashrc"
+
+# Install dependencies
+printf "Installing garage dependencies\n"
+printf "You will probably be asked for your sudo password\n"
+sudo apt-get -y -q update
+# To install garage
+sudo apt-get install -y \
+  wget \
+  bzip2 \
+  unzip \
+  git
+# For building glfw
+sudo apt-get install -y \
+  cmake \
+  xorg-dev
+# mujoco_py
+# See https://github.com/openai/mujoco-py/blob/master/Dockerfile
+# 16.04 repo is too old, install glfw from source instead of libglfw3
+sudo apt-get install -y \
+  libglew-dev \
+  patchelf \
+  libosmesa6-dev
+# OpenAI gym
+# See https://github.com/openai/gym/blob/master/Dockerfile
+sudo apt-get install -y \
+  libpq-dev \
+  ffmpeg \
+  libjpeg-dev \
+  swig \
+  libsdl2-dev
+# OpenAI baselines
+sudo apt-get install -y \
+  libopenmpi-dev \
+  openmpi-bin
+sudo apt-get clean
+sudo rm -rf /var/lib/apt/lists/*
+
+# Build GLFW because the Ubuntu 16.04 version is too old
+# See https://github.com/glfw/glfw/issues/1004
+apt-get purge -y -v libglfw*
+git clone https://github.com/glfw/glfw.git
+cd glfw
+git checkout 0be4f3f75aebd9d24583ee86590a38e741db0904
+mkdir glfw-build
+cd glfw-build
+cmake -DBUILD_SHARED_LIBS=ON -DGLFW_BUILD_EXAMPLES=OFF \
+  -DGLFW_BUILD_TESTS=OFF -DGLFW_BUILD_DOCS=OFF ..
+make -j"$(nproc)"
+make install
+cd ../../
+rm -rf glfw
+
+# Set up MuJoCo
+if [[ ! -f "${HOME}"/.mujoco/ ]]; then
+  mkdir "${HOME}"/.mujoco
+  wget https://www.roboti.us/download/mjpro150_linux.zip -O mujoco.zip
+  unzip mujoco.zip -d "${HOME}"/.mujoco
+  rm mujoco.zip
+else
+  printf '\e[0;31m%s\e[0m\n' "Mujoco is already installed!"
+fi
+# Configure MuJoCo as a shared library
+export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:${HOME}/.mujoco/mjpro150/bin"
+LD_LIB_ENV_VAR="LD_LIBRARY_PATH=\"\$LD_LIBRARY_PATH:${HOME}/.mujoco"
+LD_LIB_ENV_VAR="${LD_LIB_ENV_VAR}/mjpro150/bin\""
+if [[ "${_arg_set_envvar}" = on ]]; then
+  printf "\\n# Added by the garage installer\\n" >> "${BASH_RC}"
+  printf "export ${LD_LIB_ENV_VAR}\\n" >> "${BASH_RC}"
 fi
 
-echo "Creating conda environment..."
+# Set up conda
+hash conda 2>/dev/null || {
+  wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh \
+    -O miniconda.sh
+  sudo bash miniconda.sh -b -p /opt/conda
```

Comment:
Allow conda to put this in the default location for a user install, rather than /opt.